File: app/src/main/java/com/popcode/starwars/data/remote/ApiService.java
package com.popcode.starwars.data.remote;

import com.popcode.starwars.data.local.entity.CharacterElement;
import com.popcode.starwars.data.remote.response.CharactersResponse;
import com.popcode.starwars.data.remote.response.GenericResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiService {
    @GET("people/")
    Call<CharactersResponse> getCharacters(@Query("page") Integer page);

    @GET("people/")
    Call<CharactersResponse> findCharacter(@Query("search") String query);

    @GET("planets/{id}/")
    Call<GenericResponse> getPlanet(@Path("id") Integer id);

    @GET("species/{id}/")
    Call<GenericResponse> getSpecie(@Path("id") Integer id);

    @GET("people/{id}/")
    Call<CharacterElement> getCharacter(@Path("id") Integer id);
}

File: app/src/main/java/br/com/example/starwars/data/remote/ApiService.kt
package br.com.example.starwars.data.remote

import br.com.example.starwars.data.entities.ApiListPeopleResponse
import br.com.example.starwars.data.entities.ApiPlanet
import br.com.example.starwars.data.entities.ApiSpecie
import retrofit2.http.GET
import retrofit2.http.Query
import retrofit2.http.Url

interface ApiService {

    @GET("people")
    suspend fun getListPeople(@Query("page") page: Int): ApiListPeopleResponse

    @GET
    suspend fun getPlanet(@Url url: String): ApiPlanet

    @GET
    suspend fun getSpecie(@Url url: String): ApiSpecie
}